ninewinecfg/main.c

```diff
@@ -282,50 +314,69 @@ static void set_dlg_string(HWND hwnd, int dlg_id, int res_id)
 /*
  * Gallium nine
  */
-static BOOL nine_get(void)
+static BOOL nine_registry_enabled(void)
 {
     BOOL ret = FALSE;
     LPSTR value;
 
-    CHAR buf[MAX_PATH];
-
     if (common_get_registry_string(reg_path_dll_overrides, reg_key_d3d9, &value))
     {
         ret = !strcmp(value, reg_value_override);
         HeapFree(GetProcessHeap(), 0, value);
     }
 
+    return ret;
+}
+
+static BOOL nine_get(BOOL WasEnabled)
+{
+    CHAR buf[MAX_PATH], buf_back[MAX_PATH];
+    BOOL ret = nine_registry_enabled();
```

Comment:
This var is now used for 3 different cases. Let's keep it as return value and add `BOOL reg` and `ln` on top?

Reply:
It is return value. I just move old code to function and do the same. Check current code:
```
static BOOL nine_get(void)
{
    BOOL ret = FALSE;
    LPSTR value;

    CHAR buf[MAX_PATH];

    if (common_get_registry_string(reg_path_dll_overrides, reg_key_d3d9, &value))
    {
        ret = !strcmp(value, reg_value_override);
        HeapFree(GetProcessHeap(), 0, value);
    }
```
`LPSTR value;` and `if (common_get_registry_string(reg_path_dll_overrides, reg_key_d3d9, &value))` just goes to that function.

Reply:
I know, but in nine_get() you also assign reg and symlink values, it's less confusing if separating the various values

Reply:
Ok, I understand, that it may be confusing. But I don't understand your idea. Do you want something like:
```
static BOOL nine_get(BOOL WasEnabled)
{
    CHAR buf[MAX_PATH], buf_back[MAX_PATH];
    BOOL ret, ln;

    ln = nine_registry_enabled();
    ret = ln;
```
?

Reply:
Something like this?
```
reg = nine_registry_enabled();
ln = is_symlink(buf);
...
if (!reg && ln)
  ...
  ret = FALSE;
...
if (ln & !file_exist(buf, FALSE))
  ...
  ret = FALSE;
```

Reply:
If you want to replace `static BOOL nine_get(BOOL WasEnabled)` with `static BOOL nine_get(void)` + `ln = is_symlink(buf);` - this will not work.
`nine_get` is calling after `nine_set` and `ln = is_symlink(buf);` will get new reg state, not old one.

Reply:
Sorry, looks like I got your idea. I'll add it in next version.

Reply:
It's just cosmetic, but let's keep it easy to read ;)

Reply:
Done. I need to init `ret`, so I used `ret = reg && ln;`